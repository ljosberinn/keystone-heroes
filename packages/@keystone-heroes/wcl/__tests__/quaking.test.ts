import { graphql } from "msw";
import { setupServer } from "msw/node";

import {
  getQuakingDamageTakenEvents,
  getQuakingInterruptEvents,
} from "../src/queries/events/affixes/quaking";
import quakingDamageTakenEvents from "./fixtures/quakingDamageTakenEvents.json";
import quakingInterruptEvents from "./fixtures/quakingInterruptEvents.json";

describe("quaking", () => {
  const server = setupServer();

  beforeAll(() => {
    server.listen();
  });

  afterEach(() => {
    server.resetHandlers();
  });

  afterAll(() => {
    server.close();
  });

  test("getQuakingDamageTakenEvents", async () => {
    server.use(
      graphql.query("EventData", (_req, res, ctx) => {
        return res(
          ctx.data({
            reportData: {
              report: {
                events: {
                  data: quakingDamageTakenEvents,
                },
              },
            },
          })
        );
      })
    );

    const result = await getQuakingDamageTakenEvents({
      reportID: "",
      startTime: 0,
      endTime: 1,
    });

    expect(result).toMatchSnapshot();
  });

  test("getQuakingInterruptEvents", async () => {
    server.use(
      graphql.query("EventData", (_req, res, ctx) => {
        return res(
          ctx.data({
            reportData: {
              report: {
                events: {
                  data: quakingInterruptEvents,
                },
              },
            },
          })
        );
      })
    );

    const result = await getQuakingInterruptEvents({
      reportID: "",
      startTime: 0,
      endTime: 1,
    });

    expect(result).toMatchSnapshot();
  });
});
