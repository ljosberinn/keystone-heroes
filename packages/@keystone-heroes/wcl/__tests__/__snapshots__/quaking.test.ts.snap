// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quaking getQuakingDamageTakenEvents 1`] = `
Array [
  Object {
    "abilityGameID": 240448,
    "absorbed": 2179,
    "amount": 14342,
    "buffs": "53365.",
    "hitType": 1,
    "mitigated": 911,
    "sourceID": 5,
    "targetID": 3,
    "timestamp": 241680,
    "type": "damage",
    "unmitigatedAmount": 17432,
  },
  Object {
    "abilityGameID": 240448,
    "amount": 15929,
    "hitType": 1,
    "mitigated": 1873,
    "sourceID": 3,
    "targetID": 5,
    "timestamp": 241680,
    "type": "damage",
    "unmitigatedAmount": 17802,
  },
  Object {
    "abilityGameID": 240448,
    "amount": 16521,
    "hitType": 1,
    "mitigated": 911,
    "sourceID": 5,
    "targetID": 3,
    "timestamp": 421694,
    "type": "damage",
    "unmitigatedAmount": 17432,
  },
  Object {
    "abilityGameID": 240448,
    "amount": 15929,
    "hitType": 1,
    "mitigated": 1873,
    "sourceID": 3,
    "targetID": 5,
    "timestamp": 421694,
    "type": "damage",
    "unmitigatedAmount": 17802,
  },
  Object {
    "abilityGameID": 240448,
    "amount": 15930,
    "hitType": 1,
    "mitigated": 1872,
    "sourceID": 3,
    "targetID": 5,
    "timestamp": 961687,
    "type": "damage",
    "unmitigatedAmount": 17802,
  },
  Object {
    "abilityGameID": 240448,
    "amount": 18173,
    "buffs": "53365.",
    "hitType": 1,
    "mitigated": 1002,
    "sourceID": 5,
    "targetID": 3,
    "timestamp": 961687,
    "type": "damage",
    "unmitigatedAmount": 19175,
  },
]
`;

exports[`quaking getQuakingInterruptEvents 1`] = `
Array [
  Object {
    "abilityGameID": 240448,
    "extraAbilityGameID": 212084,
    "sourceID": 3,
    "sourceMarker": 7,
    "targetID": 3,
    "targetMarker": 7,
    "timestamp": 3870105,
    "type": "interrupt",
  },
  Object {
    "abilityGameID": 240448,
    "extraAbilityGameID": 34914,
    "sourceID": 19,
    "targetID": 19,
    "timestamp": 4190108,
    "type": "interrupt",
  },
  Object {
    "abilityGameID": 240448,
    "extraAbilityGameID": 8092,
    "sourceID": 19,
    "targetID": 19,
    "timestamp": 4310106,
    "type": "interrupt",
  },
]
`;
